Adaption for new structure

psemdel opened the pull request:
Please check it.

baronet2 commented on first_cycling_api/parser.py:
```diff
@@ -75,17 +75,32 @@ def parse_table(table):
 	# Add information hidden in tags
-	for col, series in soup_df.iteritems():
+	for col, series in soup_df.items():
+        
+        
```
Suggesting to remove these blank lines.

```suggestion
```

baronet2 commented on first_cycling_api/race/endpoints.py:
```diff
@@ -83,16 +83,12 @@ def _parse_soup(self):
-		divs = self.soup.find_all('div', {'class': "tab-content dummy"})
+		divs = self.soup.find_all('div', {'class': "tab-content"}) #includes also tab-content results
 		self.standings = {div['id']: parse_table(div.table) for div in divs}
+		self.results_table = self.standings[divs[0]['id']] #first appearing is the result
```
When I follow the first example in the README.md (for 2019 Amstel), this line is giving me the error below. Do you get this too?

```
File ~/FirstCyclingAPI/first_cycling_api/race/endpoints.py:91, in RaceEditionResults._get_results_table(self)
     89 divs = self.soup.find_all('div', {'class': "tab-content"}) #includes also tab-content results
     90 self.standings = {div['id']: parse_table(div.table) for div in divs}
---> 91 self.results_table = self.standings[divs[0]['id']]

IndexError: list index out of range
```

psemdel replied:
Ok, will check. I should have left the fallback probably.

baronet2 reviewed (commented):
Thank you for your PR @psemdel !

I tried testing your changes and had an error come up when trying to load race results. Could you please have a look and let me know if you get the same thing?

psemdel commented:
I understand what happens, they changed their system completely with the time. For instance, I changed nothing by rider, but the test fails also. I stay on the topic.

psemdel commented:
The different cases are really difficult to cover, but I found after much time a solution that seems to work for many cases (see tests). Don't hesitation to improve the code. I suppose, it is not my last commit.

baronet2 commented:
Hi @psemdel , I opened #25 which I think fixes most of the problems with a few changes. I'll wait a few days before merging in case you have any comments.

My PR does not fix the problem with the other classifications for 2023 stage races. If your PR has the solution (I haven't had time to test yet), let's rebase after merging #25. I want to handle one improvement at a time.

psemdel commented:
Sure, too many changes at once is dangerous. I had to change several things at the same time, because otherwise it worked for then new race, but not anymore for the old, and then the contrary.. So I tackled everything at once.

Compared to your PR:
* I re-inverted first and last name, as it is really an issue in my view to have then reverted. 
* I fixed the standings for new races, which was not working anymore.
* There is a check for the classification_num. Without it, it will retrieve the main result without telling us anything.

My suggestion would be that you first pick-up my supplementary tests. Then you can make up your mind from there.

I also added a handling of startlist and results (I needed it for my purposes). It is an extra feature. I really needed some effort to avoid breaking your code with it. But definitively you can add it at the end. It is the file "combi" and "combi_test". Nothing depends on them.

baronet2 commented:
Thanks @psemdel ! I just merged #25 .

I agree with your changes, and like you said let's add them one at a time. Let's do:

1. Fix the classifications/standings for 2023 stage races, and add the associated tests.
2. Add handling of startlist endpoints and associated tests.

You can convert this PR to be change 1, or you can make a new PR if that's easier.

psemdel commented:
I think, I will make a new PR tomorrow.
